drivers/ethernet/eth_adin2111.c

```diff
@@ -48,11 +50,93 @@ LOG_MODULE_REGISTER(eth_adin2111, CONFIG_ETHERNET_LOG_LEVEL);
 #define ADIN2111_UNICAST_P1_ADDR_SLOT		2U
 /* MAC Address Rule and DA Filter Port 2 slot/idx */
 #define ADIN2111_UNICAST_P2_ADDR_SLOT		3U
+/* Free slots for further filtering */
+#define ADIN2111_FILTER_FIRST_SLOT		4U
+#define ADIN2111_FILTER_SLOTS			16U
+
 /* As per RM rev. A table 3, t3 >= 50ms, delay for SPI interface to be ready */
 #define ADIN2111_SPI_ACTIVE_DELAY_MS		50U
 /* As per RM rev. A page 20: approximately 10 ms (maximum) for internal logic to be ready */
 #define ADIN2111_SW_RESET_DELAY_MS		10U
 
+int eth_adin2111_sw_reset(const struct device *dev)
+{
+	return eth_adin2111_reg_write(dev, ADIN2111_RESET, ADIN2111_RESET_SWRESET);
+}
+
+int eth_adin2111_mac_reset(const struct device *dev)
+{
```

Comment:
> Note that PHY 1 must be out of software power-down for the MAC
subsystem reset to take effect

It is worth mentioning in the function comment.